### DISCUSSION TITLE
tgcalls v1.0.0. First stable release — Telethon support, ability to implement custom MTProto bridges and more

### DISCUSSION THREAD

**Discussion by MarshalX** [MarshalX/tgcalls#108] [General]
## Introduce

**Now pytgcalls supports both the most popular MTProto clients! Pyrogram and Telethon ✨**

Starting from 1.0.0 you are able to implement your own MTProto bridges as easy as you wish!

## New installing options

Starting from installing the library, you need to choose with which client you will work.

For Pyrogram:
```bash
pip3 install pytgcalls[pyrogram]
```
For Telethon:
```bash
pip3 install pytgcalls[telethon]
```

Of course, you can manually install requires, but please pay your attention to minimal required version of these libs.

## New way to create `GroupCall` instances

Now we have one class called `GroupCallFactory` to create `GroupCall` instances of any types. You can find examples of new way in this discussion: https://github.com/MarshalX/tgcalls/discussions/101

## FAQ 

Q: Will previous code base work with new version of library?
A: Everything is individual. I made backward compatibility for the old code, which will notify the user about outdated ways of creating instances, but this will not save you from dirty hacks that could be used in your code.

Q: How to use this library with Telethon?
A: We already have a [code snippet](https://github.com/MarshalX/tgcalls/blob/main/README.md#telethons-snippet) in the main README file. Also, we have pretty and simple [examples](https://github.com/MarshalX/tgcalls/tree/main/examples) what works with Telethon too!

Q: What was done in this version?
A: In this version was optimized requests to MTProto, reworked system of connection, reconnection and disconnection. Now you can stop connection at any time and so on. Was added working with _default_join_as_ value. Was fixed several bugs.

Happy coding ❤️

<hr /><em>This discussion was created from the release <a href='https://github.com/MarshalX/tgcalls/releases/tag/v1.0.0'>tgcalls v1.0.0. First stable release — Telethon support, ability to implement custom MTProto bridges and more</a>.</em>

**Comment by john-phonk** [comment]
F*ck eleton, all my homies use pyrogram
